server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -818,4 +855,40 @@ private synchronized Runnable tryQueue(Runnable runnable) {
             return toExecute;
         }
     }
+
+    private static final class SearchOperationListenerExecutor {
+        private final SearchRequestOperationsListener listener;
+        private final SearchPhaseContext searchPhaseContext;
+
+        private final long startTime;
+
+        private long endTime;
+
+        private long tookTime;
```

Comment:
Mention time unit along with variable name. tookTimeInMillis. Applies for other as well.